src/test/skiplist_tests.cpp

```diff
@@ -37,9 +37,9 @@ BOOST_AUTO_TEST_CASE(skiplist_test)
         int from = InsecureRandRange(SKIPLIST_LENGTH - 1);
         int to = InsecureRandRange(from + 1);
 
-        BOOST_CHECK(vIndex[SKIPLIST_LENGTH - 1].GetAncestor(from) == &vIndex[from]);
-        BOOST_CHECK(vIndex[from].GetAncestor(to) == &vIndex[to]);
-        BOOST_CHECK(vIndex[from].GetAncestor(0) == &vIndex[0]);
+        BOOST_CHECK(vIndex[SKIPLIST_LENGTH - 1].GetAncestor(from) == vIndex.data() + from);
```

Comment:
well, you get the idea, if the index was other than 0, it seems to me that the change makes things worse.